crates/ty_python_semantic/src/site_packages.rs

```diff
@@ -111,23 +209,17 @@ impl PythonEnvironment {
         }
     }
 
-    /// Returns the `site-packages` directories for this Python environment,
-    /// as well as the Python version that was used to create this environment
-    /// (the latter will only be available for virtual environments that specify
+    /// Returns the Python version that was used to create this environment
+    /// (only be available for virtual environments that specify
     /// the metadata in their `pyvenv.cfg` files).
-    ///
-    /// See the documentation for [`site_packages_directory_from_sys_prefix`] for more details.
-    pub(crate) fn into_settings(
-        self,
-        system: &dyn System,
-    ) -> SitePackagesDiscoveryResult<(SitePackagesPaths, Option<PythonVersionWithSource>)> {
-        Ok(match self {
-            Self::Virtual(venv) => (venv.site_packages_directories(system)?, venv.version),
-            Self::System(env) => (env.site_packages_directories(system)?, None),
-        })
+    pub fn python_version(&self) -> Option<&PythonVersionWithSource> {
```

Comment:
I wonder if this should be called `python_version_from_metadata` since we also have `SitePackagesPaths::try_resolve_installation_python_version`, which first tries calling this method and then falls back to looking at the layout of the virtual environment on the filesystem if this method returns `None`?

Reply:
Ah I missed that! In that case, I'd possibly rename this one to `python_version_from_metadata`, and also rename `try_resolve_installation_python_version` to `python_version_from_layout`?

Reply:
Sounds good, I can do that